extensions/chocolatey-core.extension/extensions/Get-AppInstallLocation.ps1

```diff
@@ -0,0 +1,70 @@
+<#
+.SYNOPSIS
+    Get application install location
+
+.DESCRIPTION
+    Function tries to find install location in multiple places. It returns $null if all fail. The following
+    locations are tried:
+      - local and machine (x32 & x64) various Uninstall keys
+      - x32 & x64 Program Files up to the 2nd level of depth
+      - native commands available via PATH
+      - locale and machine registry key SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths
+
+    Use Verbose parameter to see which of the above locations was used for the result, if any.
+
+.EXAMPLE
+    PS> Get-AppInstallLocation choco
+
+    Returns the install location of the application 'choco'.
+
+.OUTPUTS
+    [String] or $null
+#>
+function Get-AppInstallLocation {
+    [CmdletBinding()]
+    param(
+        # Regular expression pattern
+        [ValidateNotNullOrEmpty()]
+        [string] $AppNamePattern
+    )
+
+    function strip($path) { if ($path.EndsWith('\')) { return $path -replace '.$' } else { $path } }
+
+    $ErrorActionPreference = "SilentlyContinue"
+
+    Write-Verbose "Trying local and machine (x32 & x64) Uninstall keys"
+    [array] $key = Get-UninstallRegistryKey $AppNamePattern
+    if ($key.Count -eq 1) {
```

Comment:
How it could cause a problem ? If there are two keys, it will just skip this entire section. I could select -first 1 also but the point of the function was to require no special input. 

What I did it in the past is change as soon as some package appears that it can't find but could actually find it with minor modifications. 

So lets keep it like this now, its already quite tested, and will see in the future when problem arises.